web_widget_one2many_tags/static/src/js/web_widget_one2many_tags.js

```diff
@@ -0,0 +1,93 @@
+    instance.web.form.FieldMany2ManyTags.extend({
+        initialize_texttext: function()
+        {
+            var self = this,
+                result = this._super.apply(this, arguments),
+                removeTag = result.ext.tags.removeTag;
+            result.plugins = 'tags arrow filter';
+            result.ext.core.onSetInputData = function(e, data)
+            {
+                this.input().val(data);
+            };
+            result.filter = {
+                items: []
+            };
+            result.ext.arrow = {
+                onArrowClick: function(e)
+                {
+                    var context = {},
+                        key = _.str.sprintf(
+                            'default_%s', self.field.relation_field)
+                    context[key] = self.field_manager.datarecord.id;
+                    self._search_create_popup('form', undefined, context);
+                },
+            }
+            result.ext.tags.removeTag = function(tag)
+            {
+                var id = tag.data("id"),
+                    dataset = new instance.web.DataSet(
+                        self, self.field.relation, self.build_context());
+                dataset.unlink([id]);
+                return removeTag(tag);
+            };
+            return result;
+        },
+        initialize_content: function()
+        {
+            var self = this,
+                result = this._super.apply(this, arguments);
+            if(!this.$text)
+            {
+                return result;
+            }
+            this.$text.bind('tagClick', function(e, tag, value, callback)
+            {
+                var popup = new instance.web.form.FormOpenPopup(self);
+                popup.show_element(
+                    self.field.relation, value.id, self.build_context(),
+                    {
+                        title: instance.web._t("Open: ") + value.name,
+                        readonly: self.get("effective_readonly"),
+                    });
+                popup.on('write_completed', self, function()
+                {
+                    popup.dataset.name_get(popup.dataset.ids)
+                    .then(function(names)
+                    {
+                        _(names).each(function(name)
+                        {
+                            value.name = name[1];
+                            callback(value, true);
+                        })
+                    });
+                });
+            });
+            return result;
+        },
+    });
+
+    instance.web.form.widgets.add(
+        'one2many_tags',
+        'instance.web_widget_one2many_tags.FieldOne2ManyTags'
+    );
+
+    instance.web.list.One2ManyTags = instance.web.list.Many2Many.extend({
+        init: function () {
+            this._super.apply(this, arguments);
+            // Treat it as many2many to trick odoo into populating '__display'.
+            this.type = 'many2many';
```

Comment:
Did you noticed, that this uses the default formatting if the list of related object ids is empty. I.e. it displays someting like `(0 Records)` instead of the tags. 
I was starting to write a module to remove that and display and empty string instead I was wondering if you think this could be patched into your module instead. 
I think if one2many is displayed as a list of strings aka tags, then for the empty list it would be more useful to display empty string instead of the default.

![email eingang - odoo - chromium_087](https://cloud.githubusercontent.com/assets/1388003/13708942/8af78022-e7b1-11e5-8cd7-e528e3155ab3.png)